e2e/e2etests/test_stress_eth_deposit.go

```diff
@@ -1,74 +1,255 @@
 package e2etests
 
 import (
+	"context"
 	"fmt"
 	"math/big"
+	"sync"
+	"sync/atomic"
 	"time"
 
 	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/stretchr/testify/require"
 	"github.com/zeta-chain/protocol-contracts/pkg/gatewayevm.sol"
-	"golang.org/x/sync/errgroup"
 
 	"github.com/zeta-chain/node/e2e/runner"
 	"github.com/zeta-chain/node/e2e/utils"
 	crosschaintypes "github.com/zeta-chain/node/x/crosschain/types"
 )
 
+type depositResult struct {
+	index     int
+	txHash    ethcommon.Hash
+	cctx      *crosschaintypes.CrossChainTx
+	duration  time.Duration
+	err       error
+	startTime time.Time
+}
+
 // TestStressEtherDeposit tests the stressing deposit of ether
 func TestStressEtherDeposit(r *runner.E2ERunner, args []string) {
-	require.Len(r, args, 2)
+	require.Len(r, args, 4)
 
 	// parse deposit amount and number of deposits
 	depositAmount := utils.ParseBigInt(r, args[0])
 	numDeposits := utils.ParseInt(r, args[1])
+	batchSize := utils.ParseInt(r, args[2])
+	batchInterval := utils.ParseInt(r, args[3])
 
-	r.Logger.Print("starting stress test of %d deposits", numDeposits)
+	// configure test mode
+	// true: tolerates transaction failures and reports statistics at the end
+	bestEffortMode := true
```

Comment:
should this be an argument? i see it used below but it is not configurable parameter, so it is always true?

Reply:
Right now it's a constant to easily change it in the code but I think we can change it an argument yes